### DISCUSSION TITLE
Calling plugin while using Capacitor as Script include for PWA

### DISCUSSION THREAD

**Discussion by mark-divitt** [ionic-team/capacitor#3174] [Q&A] [answered]
Hi Folks,
              Capacitor newbie here.  I am working with a PWA.  I have successfully included Capacitor Core into my webapp (via Script include), so things like this works:
  const isAvailableCamera = Capacitor.isPluginAvailable('Camera');
  console.log('Camera :' + isAvailableCamera);  // Returns true 

However, when I call this:  

 Camera.getPhoto();

It fails with:

app.js:92 Uncaught ReferenceError: Camera is not defined

also tried :       Plugins.Camera.getPhoto();  with same result

So somehow the Plugins are not being included into capacitor.js or I am referencing them wrong.

_Please note that I am not calling import anywhere, relying on script include to include plugins_

capacitor.js is minified, so rather hard to read.

Not a lot of examples for the "Script include" way of doing things.

I am not using Ionic in this case, I am using Framework7.

I did all the steps in this set of instructions : 
```
Using Capacitor as a Script Include
To use Capacitor core in a web app that is not using a build system or bundler/module loader, you must set bundledWebRuntime to true in your capacitor.config.json, tell capacitor to copy the specified version of Capacitor Core into your project, and then import capacitor.js into your index.html:

{
  "bundledWebRuntime": true
}
Copy to your project:

npx cap copy web
In index.html, import capacitor.js before your app's JS:

<script src="capacitor.js"></script>
<script src="your/app.js"></script>
```

Thanks in advance for your help!
Mark

**Comment by dotNetkow** [comment]
Hey Mark, my guess is that you're not including PWA Elements, which [is required](https://capacitorjs.com/docs/apis/camera#pwa-notes) for the Camera to work in a PWA.

Here's the [script tag details](https://capacitorjs.com/docs/pwa-elements#including-through-script-tag).

**Reply by mark-divitt** [reply]
Hi dotNetkow,
                       Thanks for the idea, however I did include those with no luck, same exact issue...

index.html:
...
  <script src="capacitor.js"></script>
  <script type="module" src="https://unpkg.com/@ionic/pwa-elements@latest/dist/ionicpwaelements/ionicpwaelements.esm.js"></script>
  <script nomodule src="https://unpkg.com/@ionic/pwa-elements@latest/dist/ionicpwaelements/ionicpwaelements.js"></script>
....

Thanks,
Mark

**Reply by dotNetkow** [reply]
Do you have a <head> tag in the index.html by chance? it's required, as Capacitor is injected there upon app initialization.

It's (very newly) mentioned in the Installation page, but I'll update the /docs/web page too).

**Reply by mark-divitt** [reply]
I do not believe so, which tag?

**Reply by dotNetkow** [reply]
Oh sorry! Looks like GitHub stripped out the head tag mention. oops! Trying to build this example out now, stay tuned.

**Comment by mark-divitt** [comment]
Ah, I see now.  My index.html does have a 
```
<head>
...lots of meta and link elements
</head>
```

**Comment by mark-divitt** [comment]
I assume you are referring to this section of the installation doc 
```
 In addition, a <head> element is needed in the main index.html file, as Capacitor is injected there upon app initialization.
```

**Comment by mark-divitt** [comment]
Thank you!

**Comment by dotNetkow** [comment, marked as answer]
OK, here's a working example in one HTML file. You can open this on the file system or via `npx cap serve` (local web server).

[The docs here](https://capacitorjs.com/docs/web#using-capacitor-as-a-script-include) are most of the way there - I will update soon but in the meantime, here's a [WIP full example](https://github.com/capacitor-community/examples/tree/vanillajs/vanilla) (subject to change!)

You must include the `capacitor.js` file locally. It can be found under `node_modules/@capacitor/core/dist/capacitor.js` or will get copied into your project automatically from there by running the `npx cap copy web` command (as the docs mention).

```
<!DOCTYPE html>
<html lang="en" dir="ltr">
<head>
  <meta charset="UTF-8">
  <title>Capacitor: Vanilla JS</title>
  <meta name="viewport" content="viewport-fit=cover, width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no">
  <meta name="format-detection" content="telephone=no">
  <meta name="msapplication-tap-highlight" content="no">

  <script type="module" src="https://unpkg.com/@ionic/pwa-elements@latest/dist/ionicpwaelements/ionicpwaelements.esm.js"></script>
  <script nomodule src="https://unpkg.com/@ionic/pwa-elements@latest/dist/ionicpwaelements/ionicpwaelements.js"></script>
  <script src="capacitor.js"></script>
  <script>
    async function takePicture() {
      const { Camera } = Capacitor.Plugins;
      const photo = await Camera.getPhoto({
          quality: 90,
          resultType: "uri"
        });
      
      const image = document.getElementById("photo");
      image.src = photo.webPath;
    }
  </script>
  </head>
  <body>
    <div id="app">
      <button type="button" onclick="takePicture();">Capture photo</button>
      <img id="photo"> 
    </div>
  </body>
</html>
```

**Reply by mark-divitt** [reply]
Excellent, thank you very much.  This helped me get this working...

**Reply by dotNetkow** [reply]
Sweet, awesome! Glad to hear. Cheers

**Reply by macsupport** [reply]
For the same use case, what would be different for Capacitor v3, since plugins are not included in capacitor.js?

**Reply by alanxp** [reply]
> would be different for Capacitor v3, since plugins are not included in capacitor.js?

Im having a nightmare with plugins not included in Capacitor 3, I think i might go back to v2
